c++/src/kj/compat/http.c++

```diff
@@ -1823,15 +1822,18 @@ private:
         } else if (amount < minBytes) {
           // We requested a minimum 1 byte above, but our own caller actually set a larger minimum
           // which has not yet been reached. Keep trying until we reach it.
-          return tryReadInternal(reinterpret_cast<byte*>(buffer) + amount,
-                                 minBytes - amount, maxBytes - amount, alreadyRead + amount);
+          buffer = reinterpret_cast<byte*>(buffer) + amount;
+          minBytes -= amount;
```

Comment:
maybe rather than mutating these it is better to use `alreadyRead` variable everywhere?

Reply:
Personally I think either way is fine, but changing things has some risk of introducing bugs.

Reply:
@mikea I'm not sure I have a picture of what you mean. Do you mean, for example, that line 1814 would become this?
```cpp
      auto amount = co_await getInner().tryRead(
          reinterpret_cast<byte*>(buffer) + alreadyRead,
          1, kj::min(maxBytes - alreadyRead, length - alreadyRead));
```